src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -293,75 +265,32 @@ public static void LoadSatelliteAssembly(byte[] dllBytes)
             }
         }
 
-        private static string GeneratedInitializerClassName = "System.Runtime.InteropServices.JavaScript.__GeneratedInitializer";
-        private static string GeneratedInitializerMethodName = "__Register_";
-
-        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Dynamic access from JavaScript")]
-        [UnconditionalSuppressMessage("Trimming", "IL2075", Justification = "Dynamic access from JavaScript")]
-        public static Task BindAssemblyExports(string? assemblyName)
+        public static unsafe Task BindAssemblyExports(string? assemblyName)
         {
-            try
-            {
-                if (string.IsNullOrEmpty(assemblyName))
-                {
-                    throw new MissingMethodException("Missing assembly name");
-                }
-                if (!assemblyName.EndsWith(".dll", StringComparison.InvariantCultureIgnoreCase))
-                {
-                    assemblyName += ".dll";
-                }
-
-                Assembly assembly = Assembly.LoadFrom(assemblyName);
-                Type? type = assembly.GetType(GeneratedInitializerClassName);
-                if (type == null)
-                {
-                    foreach (var module in assembly.Modules)
-                    {
-                        RuntimeHelpers.RunModuleConstructor(module.ModuleHandle);
-                    }
-                }
-                else
-                {
-                    MethodInfo? methodInfo = type.GetMethod(GeneratedInitializerMethodName, BindingFlags.NonPublic | BindingFlags.Static);
-                    methodInfo?.Invoke(null, []);
-                }
-
-                return Task.CompletedTask;
-            }
-            catch (Exception ex)
-            {
-                if (ex is TargetInvocationException refEx && refEx.InnerException != null)
-                    ex = refEx.InnerException;
-                return Task.FromException(ex);
-            }
+            Interop.Runtime.BindAssemblyExports(Marshal.StringToCoTaskMemUTF8(assemblyName));
+            return Task.CompletedTask;
         }
 
-        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "TODO https://github.com/dotnet/runtime/issues/98366")]
-        [UnconditionalSuppressMessage("Trimming", "IL2075", Justification = "TODO https://github.com/dotnet/runtime/issues/98366")]
         public static unsafe JSFunctionBinding BindManagedFunction(string fullyQualifiedName, int signatureHash, ReadOnlySpan<JSMarshalerType> signatures)
         {
-            if (string.IsNullOrEmpty(fullyQualifiedName))
-            {
-                throw new ArgumentNullException(nameof(fullyQualifiedName));
-            }
+            var (assemblyName, nameSpace, shortClassName, methodName) = ParseFQN(fullyQualifiedName);
```

Comment:
Since we don't need parsed values in C#. Would it make any difference to parse it in C? Or in JS?